int x
int y

float vel_x
float vel_z
